File: tests/indexing/parsing/lang/fixtures/java/Exceptions.java
class Exceptions {

    int parseOrZero(String text) {
        int value;
        try {
            value = Integer.parseInt(text);
            value = value * 2;
        } catch (NumberFormatException e) {
            value = 0;
            text = e.getMessage();
        } finally {
            text = text.trim();
            value = value + text.length();
        }
        return value;
    }

    int elementAt(int[] xs, int i, int fallback) {
        int result;
        try {
            result = xs[i];
            result = result + i;
        } catch (ArrayIndexOutOfBoundsException e) {
            result = fallback;
            fallback = fallback - 1;
        } catch (NullPointerException e) {
            result = 0;
        }
        return result;
    }

    String firstLine(Reader source) {
        String line = "";
        try (BufferedReader reader = new BufferedReader(source)) {
            line = reader.readLine();
            line = line.trim();
        } catch (IOException e) {
            line = "";
        }
        return line;
    }
}

File: tests/indexing/parsing/lang/fixtures/kotlin/Exceptions.kt
class Exceptions {

    fun parseOrZero(text: String): Int {
        var value: Int
        try {
            value = text.toInt()
            value = value * 2
        } catch (e: NumberFormatException) {
            value = 0
            value = value - 1
        } finally {
            value = value + 1
        }
        return value
    }

    fun elementAt(xs: IntArray, i: Int, fallback: Int): Int {
        var result: Int
        try {
            result = xs[i]
        } catch (e: IndexOutOfBoundsException) {
            result = fallback
            result = result - 1
        } catch (e: NullPointerException) {
            result = 0
        }
        return result
    }
}
